index.js

```diff
@@ -8,30 +8,59 @@ var Hapi = require('hapi')
 var Primus = require('primus')
 var config = require('config')
 var EventEmitter = require('eventemitter3')
-var logger = require('./lib/log')(module)
+var logger = require('./lib/util/log')(module)
 var routes = require('./lib/routes')
 var eventHandlers = require('./lib/event-handlers')
 var server = new Hapi.Server()
+var Rooms = require('primus-rooms')
 
 server.connection({
   port: config.port
 })
 
-var emitter = new EventEmitter()
-eventHandlers(emitter)
-
 var primus = Primus(server.listener, {})
+primus.use('rooms', Rooms)
 primus.on('connection', function (spark) {
   spark.write('ping')
   spark.on('data', function (data) {
-    logger.info(data)
+    if (data.room) {
+      if (data.save) {
+        // save the buffered data into the database
+        return
+      }
+      var room = data.room
+      var message = data.msg
+
+      // check if spark is already in this room
+      if (~spark.rooms().indexOf(room)) {
+        if (message && message.lineNumber && message.line) {
+          console.log(message.lineNumber + ': ' + message.line)
+          spark.room(room).write(message)
+        }
+      } else {
+        if (data.action === 'leave') {
+          spark.leave(room)
+          return
+        }
+        // join the room
+        spark.join(room, function () {
+          if (message && message.lineNumber && message.line) {
+            console.log(message.lineNumber + ': ' + message.line)
+            spark.room(room).write(message)
+          }
+        })
+      }
+    }
   })
 })
 
+var emitter = new EventEmitter()
+eventHandlers(emitter, primus)
+
 // Super simple logging
 // TODO: add something more inteligent, maybe using `good` module
 server.on('response', function (request) {
-  console.log(`${request.info.remoteAddress}: ${request.method.toUpperCase()} ${request.url.path} --> ${request.response.statusCode}`)
+  console.log(request.info.remoteAddress + ' : ' + request.method.toUpperCase() + ' ' + request.url.path + ' --> ' + request.response.statusCode)
```

Comment:
I'm going to modify the babel code to use `babel-node` instead, so we don't have that issue.